web/src/js/components/Modes/Reverse.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { ModeToggle } from "./ModeToggle";
+import Dropdown, { MenuItem } from "../common/Dropdown";
+import { useAppDispatch, useAppSelector } from "../../ducks";
+import {
+    setHost,
+    setListenConfig,
+    setProtocol,
+    toggleReverse,
+} from "../../ducks/modes/reverse";
+import ValueEditor from "../editors/ValueEditor";
+
+export default function Reverse() {
+    const dispatch = useAppDispatch();
+
+    const { active, protocol, error, listen_port, listen_host, host } =
+        useAppSelector((state) => state.modes.reverse);
+
+    const protocols = [
+        "http",
+        "https",
+        "dns",
+        "http3",
+        "quic",
+        "tcp",
+        "tls",
+        "udp",
+        "dtls",
+    ];
```

Comment:
It's super easy for this to go out of sync with the backend: https://github.com/mitmproxy/mitmproxy/blob/59fbbdc4b293cace2195c0250fcf4e7c0a808359/mitmproxy/proxy/mode_specs.py#L227.

My proposal would be to do the following:

1. Start a new PR that removes the mitmproxy version from `/state`: https://github.com/mitmproxy/mitmproxy/blob/59fbbdc4b293cace2195c0250fcf4e7c0a808359/mitmproxy/tools/web/app.py#L644
2. And then adds a backend_consts.py (open to better naming suggestions) script to https://github.com/mitmproxy/mitmproxy/tree/main/web/gen that writes the version into a TypeScript file. This "hardcodes" the mitmproxy version into the frontend so that it's available from the start.

Once this is in, we can (in this PR) extend backend_consts.py to also provide the list of supported protocols to the front end. Does that make sense?

Reply:
okay I've understood what I should do! Now, I still to figure out how since I'm not so experienced with python. But, I'll find a way and, in case, I'll ask you :)

Reply:
If you take https://github.com/mitmproxy/mitmproxy/blob/main/web/gen/state_js.py as a template, this should be super straightforward. Otherwise you can also ask @errorxyz when he's back. :)

Reply:
@mhils to generate the Typescript file and test the behaviour (for instance this: https://github.com/mitmproxy/mitmproxy/blob/main/web/gen/state_js.py), should i run the script manually? or is it somehow automated?

Reply:
The autofix CI job should pick it up.

Reply:
mhh okay, right now the tests are failing and I'm getting this: `FAILED test/mitmproxy/tools/web/test_app.py::test_generated_files[filename2] - FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/mitmproxy/mitmproxy/web/gen/../src/js/backends/consts.ts'` when running `tox -e py` since of course the file doesn't exists. If I open a PR, the autofix will run and we should solve those errors automatically, right?

Reply:
move discussions here: #7016

Reply:
okay @mhils , I'm now using the protocols that I'm generating from the backend!